### ISSUE TITLE
 Namespace 'NodeCache' has no exported member 'NodeCache'

### ISSUE THREAD

**Issue by chippawah** [joshuaslate/type-cacheable#48]
Hi and thanks for the package!

### Description

I've recently run into an issue where running `tsc` (typescript build command I'm using) is failing and pointing to `type-cacheable` as the culprit. I'm not sure if this is an issue with the package or not but I'll describe the problem and how I solved it so perhaps a PR can be discussed.

### The Error Message

```
$ tsc
node_modules/type-cacheable/dist/adapters/NodeCacheAdapter.d.ts:6:44 - error TS2694: Namespace 'NodeCache' has no exported member 'NodeCache'.

      constructor(nodeCacheClient: NodeCache.NodeCache);
                                             ~~~~~~~~~

node_modules/type-cacheable/dist/util/useAdapter.d.ts:4:62 - error TS2694: Namespace 'NodeCache' has no exported member 'NodeCache'.

  export declare const useNodeCacheAdapter: (client: NodeCache.NodeCache) => void;
                                                               ~~~~~~~~~


Found 2 errors.
```

I think it's because the type definitions in [`dist/util/useAdapter.d.ts`](https://github.com/joshuaslate/type-cacheable/blob/master/lib/util/useAdapter.ts#L11) and [`dist/adapters/NodeCacheAdapter.d.ts`](https://github.com/joshuaslate/type-cacheable/blob/master/lib/adapters/NodeCacheAdapter.ts#L15) are incorrect. `node-cache@5.0.0` doesn't have the `NodeCache.NodeCache` property anymore. I was able to fix it by just removing the second `NodeCache`.

### Updated Files

Below are the files I updated in `type-cacheable` with that usage to get it to work:

> `dist/util/useAdapter.d.ts`
```ts
import { RedisClient } from 'redis';
import * as NodeCache from 'node-cache';
export declare const useRedisAdapter: (client: RedisClient) => void;
export declare const useNodeCacheAdapter: (client: NodeCache) => void;
```

> `dist/adapters/NodeCacheAdapter.d.ts`
```ts
import * as NodeCache from 'node-cache';
import { CacheClient } from '../interfaces';
export declare class NodeCacheAdapter implements CacheClient {
    static responseCallback: (resolve: Function, reject: Function) => NodeCache.Callback<any>;
    private nodeCacheClient;
    constructor(nodeCacheClient: NodeCache);
    getClientTTL(): number;
    get(cacheKey: string): Promise<any>;
    /**
     * set - Sets a key equal to a value in a NodeCache cache
     *
     * @param cacheKey The key to store the value under
     * @param value    The value to store
     * @param ttl      Time to Live (how long, in seconds, the value should be cached)
     *
     * @returns {Promise}
     */
    set(cacheKey: string, value: any, ttl?: number): Promise<any>;
    del(cacheKey: string): Promise<any>;
}

```

### Implementation of `type-cacheable`

```ts
import * as NodeCache from "node-cache";

// Other app setup stuff

const cacheClient =  new NodeCache.default();
useNodeCacheAdapter(cacheClient);
```

### Packages Used:

| Package | Version |
| - | - |
`type-cacheable` | `1.2.1`
`node-cache` | `5.0.0`
`@types/node-cache` | `4.1.3`

Thanks again! Let me know if I need to provide more context.

**Comment by joshuaslate** [comment]
Hi @chippawah, thank you for writing a detailed issue. I released an update just now, v2.0.0. Please update and let me know if there are any lingering issues. Thank you for using type-cacheable! Have a nice day.

**Comment by chippawah** [comment]
Worked as expected! Thanks @joshuaslate !
